sdk/daml-lf/engine/src/test/scala/com/digitalasset/daml/lf/engine/ValueTranslatorSpec.scala

```diff
@@ -429,29 +496,65 @@ class ValueTranslatorSpec
             typ shouldBe TRecordUpgradable
           },
         ),
+        (
+          TRecordUpgradable,
+          ValueRecord(
+            "",
+            ImmArray(), // missing a non-optional field
+          ),
+          { case Error.Preprocessing.TypeMismatch(typ, _, _) =>
+            typ shouldBe TRecordUpgradable
+          },
+        ),
         (
           TRecordUpgradable,
           ValueRecord(
             "",
             ImmArray(
               "fieldA" -> aInt,
-              "fieldB" -> someParty, // Here the field has type Party instead of Text
```

Comment:
Is there a reason we're dropping this test of field mismatches to create a test of an extra non-optional field? Couldn't we make the non-optional field test a wholly new test instead?

Reply:
This test case was an accidental duplicate of the same test case on line 403.